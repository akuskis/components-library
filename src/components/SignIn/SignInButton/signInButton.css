.sign-in-button {
  width: 100%;
}
