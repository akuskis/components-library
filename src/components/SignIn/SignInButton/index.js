// @flow

import { Button } from '@material-ui/core';
import React from 'react';

import './signInButton.css';

type Props = {
  onClick: Function
};

const SignInButton = ({ onClick }: Props) => (
  <Button name="submit" className="sign-in-button" onClick={onClick}>
    ENTER
  </Button>
);

export default SignInButton;
