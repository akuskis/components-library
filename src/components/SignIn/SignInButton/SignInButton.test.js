import React from 'react';
import { shallow } from 'enzyme';

import SignInButton from '.';

describe('<SignInButton />', () => {
  it('should render without props', () => {
    const shallowed = shallow(<SignInButton />);

    expect(shallowed).toMatchSnapshot();
  });

  it('should render with props', () => {
    const shallowed = shallow(<SignInButton onClick={jest.fn()} />);

    expect(shallowed).toMatchSnapshot();
  });

  it('should call a function on click', () => {
    const cbFunction = jest.fn();
    const component = shallow(<SignInButton onClick={cbFunction} />);

    expect(cbFunction).toHaveBeenCalledTimes(0);

    component.simulate('click');

    expect(cbFunction).toHaveBeenCalledTimes(1);
  });
});
