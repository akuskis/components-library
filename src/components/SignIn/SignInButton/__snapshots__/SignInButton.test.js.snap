// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInButton /> should render with props 1`] = `
<WithStyles(Button)
  className="sign-in-button"
  name="submit"
  onClick={[Function]}
>
  ENTER
</WithStyles(Button)>
`;

exports[`<SignInButton /> should render without props 1`] = `
<WithStyles(Button)
  className="sign-in-button"
  name="submit"
>
  ENTER
</WithStyles(Button)>
`;
